File: CrmApp.Domain/Entities/DictionariesElements.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.Entities
{
    public class DictionariesElements
    {
        [Key]
        public int Id { get; set; }
        public int DictionaryId { get; set; }
        public string Key { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public int? CreatedBy { get; set; }
        public bool IsCustom { get; set; }
        public bool IsActive { get; set; }
        public bool IsDefault { get; set; }
        public int? ParentId { get; set; }
        public int OrdinalNumber { get; set; }
        public string? AlternativeValuesForMapping { get; set; }
        public string? Icon { get; set; }
        public string? IconColor { get; set; }

        [ForeignKey(nameof(DictionaryId))]
        public virtual Dictionaries? Dictionary { get; set; }

        [ForeignKey(nameof(CreatedBy))]
        public virtual UsersProfiles? CreatedByUser { get; set; }
    }
}

File: CrmApp.Client/src/models/DictionariesElements.ts
import { Dictionaries } from "./Dictionaries";
import { UsersProfiles } from "./UsersProfiles";

export class DictionariesElements {
    id: number = 0;
    dictionaryId: number = 0;
    key: string = "";
    value: string = "";
    createdAt: Date = new Date();
    createdBy?: number;
    isCustom: boolean = true;
    isActive: boolean = true;
    isDefault: boolean = false;
    parentId?: number;
    ordinalNumber: number = 0;
    alternativeValuesForMapping?: string;
    icon?: string;
    iconColor?: string;

    dictionary?: Dictionaries;
    createdByUser?: UsersProfiles;
}